src/components/common/SearchFiltersWrapperV2/index.tsx

```diff
@@ -98,15 +111,57 @@ export default function SearchFiltersWrapperV2({
     [currentFilters, iconFilters, featuredFilters, setCurrentFilters]
   );
 
+  useMemo(() => {
+    if (stickyFilters && Object.keys(sessionFilters).length > 0 && Object.keys(currentFilters).length === 0) {
+      const sessionFiltersToApply: Record<string, SearchNodePayload> = {};
+      const existingKeys = Object.keys(sessionFilters);
+      existingKeys?.forEach((key) => {
+        const valuesOfTheExistingKey = Array.isArray(sessionFilters[key])
+          ? (sessionFilters[key] as any)
+              .map((val: string) => val?.toString() || null)
+              .filter((val: string | null) => val !== null)
+          : [];
+
+        sessionFiltersToApply[key] = {
+          field: key,
+          operation: 'field',
+          type: 'Exact',
+          values: valuesOfTheExistingKey,
+        };
+      });
+
+      if (Object.keys(sessionFiltersToApply).length > 0) {
+        setCurrentFilters({
+          ...sessionFiltersToApply,
+        });
+      }
+    }
+  }, [sessionFilters, currentFilters, stickyFilters]);
```

Comment:
This should be a `useEffect` since no value is being returned from the `useMemo`:

```suggestion
  useEffect(() => {
    if (stickyFilters && Object.keys(sessionFilters).length > 0 && Object.keys(currentFilters).length === 0) {
      const sessionFiltersToApply: Record<string, SearchNodePayload> = {};
      const existingKeys = Object.keys(sessionFilters);
      existingKeys?.forEach((key) => {
        const valuesOfTheExistingKey = Array.isArray(sessionFilters[key])
          ? (sessionFilters[key] as any)
              .map((val: string) => val?.toString() || null)
              .filter((val: string | null) => val !== null)
          : [];

        sessionFiltersToApply[key] = {
          field: key,
          operation: 'field',
          type: 'Exact',
          values: valuesOfTheExistingKey,
        };
      });

      if (Object.keys(sessionFiltersToApply).length > 0) {
        setCurrentFilters({
          ...sessionFiltersToApply,
        });
      }
    }
  }, [sessionFilters, currentFilters, stickyFilters]);
```

Reply:
Yes, I think I was getting an infinite re-render with useEffect and that's why I changed it, but I just tested it again with useEffect and it is woking correctly now